modules/gatk4/splitintervals/main.nf

```diff
@@ -0,0 +1,47 @@
+process GATK4_SPLITINTERVALS {
+    tag "$meta.id"
+    label 'process_low'
+
+    conda (params.enable_conda ? "bioconda::gatk4=4.2.6.1" : null)
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/gatk4:4.2.6.1--hdfd78af_0':
+        'quay.io/biocontainers/gatk4:4.2.6.1--hdfd78af_0' }"
+
+    input:
+    tuple val(meta), path(intervals)
+    path(fasta)
+    path(fasta_fai)
+    path(dict)
+
+    output:
+    tuple val(meta), path("**.interval_list"), emit: split_intervals
+    path "versions.yml"           , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    def reference = fasta ? "--reference $fasta" : ""
+
+    def avail_mem = 3
+    if (!task.memory) {
+        log.info '[GATK SplitIntervals] Available memory not known - defaulting to 3GB. Specify process memory requirements to change this.'
+    } else {
+        avail_mem = task.memory.giga
+    }
+
+    """
+    gatk --java-options "-Xmx${avail_mem}g" SplitIntervals \\
+        --output ${prefix} \\
+        --intervals $intervals \\
+        $reference \\
+        $args
```

Comment:
add `--tmp-dir . \\` We had weird effects in the past when telling nextflow to write tmp files elsewhere and GTAK tools stil writing to `/tmp`, so decided to set this always.